activejob/lib/active_job/enqueuing.rb

```diff
@@ -34,7 +34,60 @@ def perform_later(...)
         enqueue_result
       end
 
+      # Push many jobs onto the queue at once. The parameter +multi_args+ is an array of arrays or hashes.
+      # Each item in the array contains the arguments for a single job. For example, the following arguments
+      # would enqueue two jobs with three arguments each.
+      #
+      # +multi_args+: [[1, "String arg", <#SomeModel ...>], [2, "String arg", <#SomeModel ...>]]
+      # First job enqueued with [1, "String arg", <#SomeModel ...>]
+      # Second job enqueued with [2, "String arg", <#SomeModel ...>]
+      #
+      # For a job with keyword arguments, the arguments can be passed as a hash:
+      #
+      # +multi_args+: [{ foo: "bar", number: 1 }, { foo: "baz", number: 2 }]
+      #
+      # Returns the number of jobs successfully enqueued.
+      def perform_all_later(multi_args, options: {})
+        if queue_adapter.respond_to?(:enqueue_all)
+          jobs = []
+          serialized_jobs = []
+
+          multi_args.each do |args|
+            job = job_from_multi_args(args).set(options)
+            job.run_callbacks(:enqueue, :before) do
+              job.run_callbacks(:enqueue, :around) do
+                jobs << job
+                serialized_jobs << job.serialize
+              end
+            end
+          end
+
+          result = queue_adapter.enqueue_all(jobs, serialized_jobs)
```

Comment:
It looks like we’re not handling `scheduled_at` here. Does Sidekiq support bulk-enqueuing scheduled jobs? Do we need a separate `enqueue_all_at` adapter method?